src/chunk_constraint.c

```diff
@@ -546,13 +573,26 @@ chunk_constraint_create_on_chunk(Chunk *chunk, Oid constraint_oid)
 							chunk->fd.hypertable_id);
 }
 
+typedef struct HypertableConstraintOperation
+{
+	const char *hypertable_constraint_name;
+} HypertableConstraintOperation;
+
 typedef struct DeleteConstraintInfo
 {
+	const char *hypertable_constraint_name;
 	int32		chunk_id;
 	Oid			chunk_oid;
-	char	   *hypertable_constraint_name;
 } DeleteConstraintInfo;
 
+typedef struct RenameHypertableConstraintInfo
+{
+	const char *hypertable_constraint_name;
+	int32		chunk_id;
```

Comment:
Also in this struct the `chunk_id` seems unnecessary since the `chunk_id` is also in read from the tuple in `chunk_constraint_rename_hypertable_tuple()`. You also removed the extra `chunk_id` check in ` hypertable_constraint_tuple_filter()`

Reply:
fixed